LeaderboardBackend/Services/Impl/LeaderboardService.cs

```diff
@@ -42,4 +43,34 @@ public async Task<CreateLeaderboardResult> CreateLeaderboard(CreateLeaderboardRe
 
         return lb;
     }
+
+    public async Task<RestoreLeaderboardResult> RestoreLeaderboard(long id)
+    {
+        Leaderboard? lb = await applicationContext.Leaderboards.FindAsync([id]);
+
+        if (lb == null)
+        {
+            return new LeaderboardNotFound();
+        }
+
+        if (lb.DeletedAt == null)
+        {
+            return new LeaderboardNeverDeleted();
+        }
+
+        Leaderboard? maybe = await applicationContext.Leaderboards.SingleOrDefaultAsync(board => board.Slug == lb.Slug && board.DeletedAt == null);
+
+        if (maybe != null)
+        {
+            return new RestoreLeaderboardConflict(maybe);
+        }
+
+        applicationContext.Leaderboards.Update(lb);
```

Comment:
I don't think this is necessary. You already fetched it so it should track automatically.

Reply:
Oh yeah you're right